src/compile/selection/interval.ts

```diff
@@ -135,6 +135,7 @@ const interval:SelectionCompiler = {
 
     return [{
       type: 'rect',
+      clip: true,
```

Comment:
Not really the main point of this PR, but it may be worth giving a name to this rect mark too?   (That will help skimming at the diff spec a lot.)